modules/settings/module.php

```diff
@@ -173,16 +155,46 @@ public function on_connect(): void {
 	public function save_plan_data( $register_response ) : void {
 		if ( $register_response && ! is_wp_error( $register_response ) ) {
 			$decoded_response = json_decode( $register_response );
-			Data::set_subscription_id( $decoded_response->id );
+			Data::set_subscription_id( $decoded_response->plan->subscription_id );
 			update_option( Settings::PLAN_DATA, $decoded_response );
 			update_option( Settings::IS_VALID_PLAN_DATA, true );
 			$this->set_default_settings();
+            self::set_plan_data_refresh_transient();
 		} else {
 			Logger::error( esc_html( $register_response->get_error_message() ) );
 			update_option( Settings::IS_VALID_PLAN_DATA, false );
 		}
 	}
 
+	public static function refresh_plan_data() {
+		$plan_data = Settings::get( Settings::PLAN_DATA );
+
+        // Refresh only if refresh transient is expired
+        if ( self::get_plan_data_refresh_transient() ) {
+            return;
+        }
+
+        // Return if plan data does not exist
+        if ( ! $plan_data ) {
+            return;
+        }
```

Comment:
Indentation is off here
and Im pretty sure that if you exit if you don't have plan data you will never fetch it....